src/test/java/bio/terra/service/search/SearchServiceTest.java

```diff
@@ -0,0 +1,131 @@
+package bio.terra.service.search;
+
+import bio.terra.common.category.Unit;
+import bio.terra.model.SearchIndexRequest;
+import bio.terra.service.resourcemanagement.google.GoogleProjectResource;
+import bio.terra.service.snapshot.Snapshot;
+import bio.terra.service.snapshot.SnapshotTable;
+import bio.terra.service.tabulardata.google.BigQueryPdao;
+import org.elasticsearch.client.IndicesClient;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.client.indices.CreateIndexRequest;
+import org.elasticsearch.client.indices.CreateIndexResponse;
+import org.elasticsearch.client.indices.GetIndexRequest;
+import org.elasticsearch.client.indices.GetIndexResponse;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.mockito.Mockito.when;
+
+@Category(Unit.class)
+public class SearchServiceTest {
+    private static final String sqlQuery = "SELECT GENERATE_UUID() uuid, CURRENT_TIMESTAMP() as now" +
+        " FROM UNNEST(GENERATE_ARRAY(1, 3));";
+
+    private static final String indexName = "idx-mock";
+
+    @Mock
+    private BigQueryPdao bigQueryPdao;
+
+    @Mock
+    private RestHighLevelClient client;
+
+    @Mock
+    private IndicesClient indicesClient;
+
+    private SearchService service;
+
+    private SearchIndexRequest searchIndexRequest;
+    private Snapshot snapshot;
+    private List<Map<String, Object>> values;
+
+    @Before
+    public void setup() throws Exception {
+        MockitoAnnotations.openMocks(this);
+        service = new SearchService(bigQueryPdao, client);
+
+        searchIndexRequest = getSearchIndexRequest();
+        snapshot = getSnapshot();
+        values = getSnapshotTableData();
+    }
+
+    @Test
+    public void indexSnapshotTest() throws Exception {
+        mockGetSnapshotTableData();
+        mockIndexRequest();
+        mockIndexResponse();
+        service.indexSnapshot(snapshot, searchIndexRequest);
+    }
```

Comment:
I think a test needs at least one `assert()`. Is there something that can be asserted here to ensure that code did something? If all we can do is assert that some API was called with some arguments, you can use `Mockito.verify()` to do that.

Reply:
Good point. I think I can assert the response.

Reply:
I added this.